pkg/clustermanager/cluster_manager.go

```diff
@@ -833,6 +838,21 @@ func (c *ClusterManager) waitForAllControlPlanes(ctx context.Context, cluster *t
 	return nil
 }
 
+func (c *ClusterManager) removeOldMachineDeployments(ctx context.Context, workloadCluster *types.Cluster, provider providers.Provider, currentSpec, newSpec *cluster.Spec) error {
+	workerConfigs := provider.BuildMapForWorkerNodeGroupsByName(newSpec.Spec.WorkerNodeGroupConfigurations)
```

Comment:
curious why does this need to come from the provider?
Are you thinking the provider might have some hidden logic that you want to abstract here?

Asking because it seems that this is only creating a set of strings and we are using to speed up the "search" of deleted worker node groups.

It would be different if the provider returned the name of the machine deployments.

Reply:
Yes, good point. Nothing provider-specific is happening here so it doesn't *need* to come from the provider. I added it because we need to access this method from the provider & cluster manager, so I added it here avoid redefining this method in cluster manger since we already have access to the provider.